benches/cub/device/merge_sort/custom_less_thrust_comparison.cu

```diff
@@ -0,0 +1,85 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+#include <thrust/random.h>
+#include <thrust/shuffle.h>
+
+#include <thrust/detail/temporary_array.h>
+
+#include <cub/device/device_merge_sort.cuh>
+
+/*
+ * Arithmetic types with the custom comparator lead to merge sort usage.
+ */
+
+class less_comparator
+{
+public:
+  template <typename T>
+  __device__ bool operator()(T i, T j) noexcept
+  {
+    return i < j;
+  }
+};
+
+template <typename T,typename Derived>
+thrust::detail::temporary_array<T, Derived> create_tmp_storage(
+  thrust::execution_policy<Derived> &policy, std::size_t size)
+{
+  return thrust::detail::temporary_array<T, Derived>(policy, size);
+}
+
+template <typename T>
+void custom_less(nvbench::state &state, nvbench::type_list<T>)
+{
+  const auto elements = static_cast<long long int>(state.get_int64("Elements"));
+
+  thrust::device_vector<T> data(elements);
+  thrust::sequence(data.begin(), data.end());
+
+  thrust::default_random_engine rng;
+
+  state.add_element_count(elements);
+  state.add_global_memory_reads<T>(elements);
+  state.add_global_memory_writes<T>(elements);
+
+  size_t temp_size = 0;
+  cub::DeviceMergeSort::SortKeys(
+    nullptr,
+    temp_size,
+    thrust::raw_pointer_cast(data.data()),
+    elements,
+    less_comparator());
+
+  using namespace nvbench::exec_tag;
+  state.exec(timer | sync, // This benchmark needs a timer and syncs internally
+             [&](nvbench::launch &launch, auto &timer) {
+               auto policy = thrust::device.on(launch.get_stream());
+               thrust::shuffle(policy, data.begin(), data.end(), rng);
+               timer.start();
+
+               // Shuffle uses temporary storage, so we have to reuse this
+               // memory in SortKeys, just like thrust does. Otherwise, we
+               // won't fit in the memory for float64.
+               auto tmp = create_tmp_storage<thrust::detail::uint8_t>(policy, temp_size);
```

Comment:
Do we need to include tmp_storage allocation in the timed critical section?

Reply:
That's how it's implemented in the thrust version. Otherwise, we won't get an honest comparison.

Reply:
I see. In that case, I don't think we need this benchmark in the regression suite, since it's just a check to compare performance before/after moving the kernel. We can keep it in a side branch for testing until you don't need it anymore.